src/http/web_socket.cr

```diff
@@ -97,6 +97,8 @@ class HTTP::WebSocket
   def close(message = nil)
     return if closed?
     @closed = true
+
+    @on_close.try &.call(message || "")
```

Comment:
This is a change of behaviour, previously `@on_close` would only fire when a websocket close frame was sent from the remote side. I'm not sure this is good change.

Reply:
That's correct. I'm not sure why wouldn't it be a good change, since you'd expect to have it called either way, IMHO it should be like that from the beginning.

Reply:
> since you'd expect to have it called either way

Personally I wouldn't expect *any* callbacks to be called after calling `websocket.close`, let alone `on_close`. Not sure what others think.

Reply:
`close` is an explicit call from user code. There's no need for a callback to notify it was called.

Reply:
I've changed it back.